pkg/subscriptions/subscriptions-view.jsx

```diff
@@ -0,0 +1,247 @@
+        // only consider primary mouse button
+        if (!e || e.button !== 0)
+            return;
+        this.props.register();
+        e.stopPropagation();
+    },
+    handleUnregisterSystem: function(e) {
+        // only consider primary mouse button
+        if (!e || e.button !== 0)
+            return;
+        this.props.unregister();
+        e.stopPropagation();
+    },
+    render: function() {
+        var errorMessage;
+        if (this.props.error) {
+            errorMessage = (
+                <DismissableError dismissError={this.props.dismissError}>{this.props.error}</DismissableError>
+            );
+        }
+
+        var label;
+        var action;
+        var note;
+        var isUnregistering = (this.props.status == "unregistering");
+        if (this.props.status == 'Unknown') {
+            label = <label>{ _("Status: System isn't registered") }</label>;
+            action = (<button title="Register"
+                              className="btn btn-primary"
+                              onClick={this.handleRegisterSystem}>{_("Register")}</button>
+            );
+        } else {
+            label = <label>{ cockpit.format(_("Status: $0"), this.props.status) }</label>;
+            action = (<button title="Unregister"
+                              className="btn btn-primary" disabled={isUnregistering}
+                              onClick={this.handleUnregisterSystem}>{_("Unregister")}</button>
+            );
+            if (isUnregistering) {
+                note = (
+                    <div className="dialog-wait-ct">
+                        <div className="spinner spinner-sm"></div>
+                        <span>{ _("Unregistering system...") }</span>
+                    </div>
+                );
+            }
+        }
+        return (
+            <div className="subscription-status-ct">
+                <h2>{_("Subscriptions")}</h2>
+                {errorMessage}
+                {label}
+                {action}
+                {note}
+            </div>
+        );
+    }
+});
+
+/* Show subscriptions status of the system and registered products, offer to register/unregister the system
+ * Expected properties:
+ * status       subscription status
+ * error        error message to show (in EmptyState if not connected, as a dismissable alert otherwise
+ * dismissError callback, triggered for the dismissable error in connected state
+ * products     subscribed products (properties as in subscriptions-client)
+ * register     callback, triggered when user clicks on register
+ * unregister   callback, triggered when user clicks on unregister
+ */
+var SubscriptionsPage = React.createClass({
+    render: function() {
+        var self = this;
+        if (this.props.status === undefined ||
+            this.props.status == 'not-found' || this.props.status == 'access-denied') {
+            var icon;
+            var description;
+            var message;
```

Comment:
Variable hosting here. These should be declared outside of the if block, or it should be separated out into a helper function.